File: backend/src/main/java/com/jozwiaka/backend/entity/Tree.java
package com.jozwiaka.backend.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "tree")
public class Tree {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

//    @OneToMany(mappedBy = "tree", cascade = CascadeType.ALL)
//    private List<Skill> skills;

    @Column(unique = true, nullable = false)
    private String name;
}

File: frontend/src/app/common/entities/tree.ts
import { Skill } from './skill';

export class Tree {
  skills: Skill[] = [];
  points: number = 0;
  tier1: number = 1;
  tier2: number = 2;
  tier3: number = 3;
  constructor(public id: number = -1, public name: string = '') {}
}
